### PR TITLE
Return deleted data provider descriptor and related swagger doc updates

### PR DESCRIPTION
**Pull request by bhufmann**
- server: Return derived data provider descriptor when deleting it.
Update swagger description for deletion and method name.
- swagger: Fix response description for successful DP creation.
Update also method name.

Signed-off-by: Bernd Hufmann <bernd.hufmann@ericsson.com>

### CODE REVIEW

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java [hunk 1]
```diff
@@ -1326,11 +1326,11 @@ public Response createDataProvider(
     @Tag(name = OCG)
     @Produces(MediaType.APPLICATION_JSON)
     @Operation(summary = "Delete a configuration instance of a given configuration type", responses = {
-            @ApiResponse(responseCode = "200", description = "The derived data provider (and it's configuration) was successfully deleted", content = @Content(schema = @Schema(implementation = org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.Configuration.class))),
+            @ApiResponse(responseCode = "200", description = "Returns the deleted derived data provider descriptor. The derived data provider (and it's configuration) was successfully deleted.", content = @Content(schema = @Schema(implementation = DataProvider.class))),
```
Comment (PatrickTasse) [inline trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java:1329]: its

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java [hunk 2]
```diff
@@ -1326,11 +1326,11 @@ public Response createDataProvider(
     @Tag(name = OCG)
     @Produces(MediaType.APPLICATION_JSON)
     @Operation(summary = "Delete a configuration instance of a given configuration type", responses = {
```
Comment (PatrickTasse) [inline trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java:1328]: Delete a derived data provider.

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java [hunk 3]
```diff
@@ -1326,11 +1326,11 @@ public Response createDataProvider(
     @Tag(name = OCG)
```
Comment (PatrickTasse) [inline trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java:1326]: At line 1314, replace 'from created by' with 'created from'?
Also check the param and return tags.

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java (hunk 1 above)
Comment (bhufmann) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java:1329]: Done

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java (hunk 2 above)
Comment (bhufmann) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java:1328]: Done

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java (hunk 3 above)
Comment (bhufmann) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java:1326]: Done

**Review by PatrickTasse — DISMISSED** [review]

**Review by PatrickTasse — APPROVED** [review]

**Pull request merged**